src/utils/transactions.ts

```diff
@@ -16,27 +16,90 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import {
+  AssetAmount,
+  calcTxAmountsDeltaForAddress,
+  convertToNegative,
+  getDirection,
+  isConsolidationTx,
+  isSwap,
+  TransactionDirection,
+  TransactionInfo,
+  TransactionInfoType
+} from '@alephium/sdk'
+import { ALPH } from '@alephium/token-list'
 import { explorer } from '@alephium/web3'
 
+import { store } from '~/store/store'
 import { Address } from '~/types/addresses'
 import { AddressPendingTransaction, AddressTransaction } from '~/types/transactions'
 
-// TODO: Same as in desktop wallet, move to SDK?
-export const extractNewTransactionHashes = (
-  incomingTransactions: explorer.Transaction[],
-  existingTransactions: explorer.Transaction['hash'][]
-): explorer.Transaction['hash'][] =>
-  incomingTransactions
-    .filter((newTx) => !existingTransactions.some((existingTx) => existingTx === newTx.hash))
-    .map((tx) => tx.hash)
-
 export const isPendingTx = (tx: AddressTransaction): tx is AddressPendingTransaction =>
   (tx as AddressPendingTransaction).status === 'pending'
 
 // TODO: Same as in desktop wallet, move to SDK?
-export const getTransactionsOfAddress = (transactions: explorer.Transaction[], address: Address) =>
-  transactions.filter(
-    (tx) =>
-      tx.inputs?.some((input) => input.address === address.hash) ||
-      tx.outputs?.some((output) => output.address === address.hash)
-  )
+export const getTransactionInfo = (tx: AddressTransaction, showInternalInflows?: boolean): TransactionInfo => {
+  const state = store.getState()
+  const assetsInfo = state.assetsInfo.entities
+  const addresses = Object.values(state.addresses.entities) as Address[]
+
+  let amount: bigint | undefined = BigInt(0)
+  let direction: TransactionDirection
+  let infoType: TransactionInfoType
+  let outputs: explorer.Output[] = []
+  let lockTime: Date | undefined
+  let tokens: Required<AssetAmount>[] = []
+
+  if (isPendingTx(tx)) {
+    direction = 'out'
```

Comment:
Note: we will need to make this more flexible (cf. explorer) as pending TXs could be incoming as well.